go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -385,3 +405,18 @@ func (oc *BaseSecondaryLayer2NetworkController) syncNodes(nodes []interface{}) e
 
 	return nil
 }
+
+func (oc *BaseSecondaryLayer2NetworkController) syncIPAMClaims(ipamClaims []interface{}) error {
+	switchType := types.OVNLayer2Switch
+	if oc.NetInfo.TopologyType() == types.LocalnetTopology {
+		switchType = types.OVNLocalnetSwitch
+	}
+	if oc.allocatesPodAnnotation() {
+		return oc.ipamClaimsReconciler.Sync(ipamClaims, oc.lsManager.ForSwitch(oc.GetNetworkScopedName(switchType)))
+	}
+	return nil
+}
+
+func (oc *BaseSecondaryLayer2NetworkController) allowPersistentIPs() bool {
+	return oc.NetInfo.AllowsPersistentIPs() && util.DoesNetworkRequireIPAM(oc.NetInfo)
+}
```

Comment:
I would just have this method check for `oc.NetInfo.AllowsPersistentIPs()` as well. Also it should check for the proper topologies as you do in the interconnect side (although it is also checked within the NetInfo, but just for consistency)

I think it should live in the `BaseSecondaryNetworkController` as I think you need to check it on the IP release code too.

Reply:
Are you saying I should move all logic to `oc.NetInfo.AllowsPersistentIPs()` ? 

I just wanted to have it simply expose the allows persistent ips attribute, and compute if the network can actually support the feature elsewhere. I think those are 2 different things. 

But honestly, you're the maintainer, let me know what you want. 

It does make sense to check here for the topology here though; I'll do that.

Also moved it to `BaseSecondaryNetworkController`.

Reply:
Just have the same you have here:
```
func (ncc *networkClusterController) allowPersistentIPs() bool {
	return ncc.NetInfo.AllowsPersistentIPs() &&
		util.DoesNetworkRequireIPAM(ncc.NetInfo) &&
		(ncc.NetInfo.TopologyType() == types.Layer2Topology || ncc.NetInfo.TopologyType() == types.LocalnetTopology)
}
```

Reply:
Isn't that what I've done ? 
https://github.com/ovn-org/ovn-kubernetes/pull/4165/commits/8508eb082678d3cd3bdcde2a829972ac69663b41

Reply:
Not sure, I have not checked your latest push. Just answering your question.